pkg/api/controller.go

```diff
@@ -4637,11 +4637,14 @@ func (c *Controller) MergeIntoBranch(w http.ResponseWriter, r *http.Request, bod
 		swag.StringValue(body.Strategy),
 		graveler.WithForce(swag.BoolValue(body.Force)))
 
+	var v *graveler.HookAbortError
```

Comment:
Shouldn't this logic be in the action execution points in graveler?

Reply:
I don't think graveler should be aware of what happens in a hook, it should be opaque. The meaning of an execution error vs an explicit failure is (imo) out of scope. I chose to put it here because the main "benefit" of this capability is how the error is exposed, so the API (presentation) layer felt like the natural choice.

Reply:
Thanks - so why are we doing it only or the Merge flow?

Reply:
Actually I wasn't :) turns out errors are already handled in `handleAPIError` including HookAbortError - so I went ahead and removed this.